src/Compilers/CSharp/Portable/Syntax/SyntaxNormalizer.cs

```diff
@@ -391,6 +397,118 @@ private static int LineBreaksAfterSemicolon(SyntaxToken currentToken, SyntaxToke
             }
         }
 
+        private static bool? TryNeedsSeparatorForPropertyPattern(SyntaxToken token, SyntaxToken next)
+        {
+            PropertyPatternClauseSyntax? propPattern =
+                token.Parent as PropertyPatternClauseSyntax ??
+                next.Parent as PropertyPatternClauseSyntax;
```

Comment:
can you do this as kind checks first.  type checks can be expensive, and based on how this function is shaped, it looks like you do this for all pairs of tokens?

Reply:
I changed it to IsKind calls...

Reply:
Neal and @agocke did benchmarks on this a year or two ago, and it actually was not super significant (half a nanosecond or so, which basically vanished when you actually do the cast afterwards).

Reply:
I'd prefer, for this `if/else` chain, we just did a switch statement with type tests and return in default.

Reply:
Actually, I suppose it would be cleaner to just do what you had originally, with the `as` expressions. Not super important if you want to leave it as is, but with only two `as` type tests it's not a perf concern.